sensor/common/compliance/nodescan_handler_impl.go

```diff
@@ -43,40 +64,46 @@ func (c *nodeScanHandlerImpl) ProcessMessage(_ *central.MsgToSensor) error {
 	return nil
 }
 
-func (c *nodeScanHandlerImpl) run() {
-	defer c.stoppedC.SignalWithError(c.stopC.Err())
-	defer close(c.toCentral)
-
-	for !c.stopC.IsDone() {
-		select {
-		case <-c.stopC.Done():
-			return
-		case scan, ok := <-c.nodeScans:
-			if !ok {
-				c.stopC.SignalWithError(errors.New("channel receiving node scans v2 is closed"))
+// run handles the messages from Compliance and forwards them to Central
+// This is the only goroutine that writes into the toCentral channel, thus it is responsible for creating and closing that chan
+func (c *nodeScanHandlerImpl) run() <-chan *central.MsgFromSensor {
+	toC := make(chan *central.MsgFromSensor)
+	go func() {
+		defer func() {
+			c.stoppedC.SignalWithError(c.stopC.Err())
+		}()
+		defer close(toC)
+		for !c.stopC.IsDone() {
+			select {
+			case <-c.stopC.Done():
 				return
+			case scan, ok := <-c.nodeScans:
+				if !ok {
+					c.stopC.SignalWithError(errInputChanClosed)
+					return
+				}
```

Comment:
When I run tests in GoLand with coverage, they show that this code branch isn't covered. I think it's significant enough to be tested.
![image](https://user-images.githubusercontent.com/537715/204329450-c8614410-a7f0-440b-93f0-88a09156aaec.png)